src/coreclr/jit/assertionprop.cpp

```diff
@@ -3118,23 +3118,52 @@ GenTree* Compiler::optVNConstantPropOnTree(BasicBlock* block, GenTree* tree)
         }
         break;
 
-#if FEATURE_HW_INTRINSICS
+#if FEATURE_SIMD
         case TYP_SIMD8:
+        {
+            simd8_t value = vnStore->ConstantValue<simd8_t>(vnCns);
+
+            GenTreeVecCon* vecCon = gtNewVconNode(tree->TypeGet(), CORINFO_TYPE_FLOAT);
```

Comment:
We talked a little bit about this offline regarding CORINFO_TYPE_FLOAT here, I would probably put a comment about why we are doing it.

Reply:
I'm going to log an issue and hold off on doing this for two reasons:
1. It avoids churning CI again for a comment change
2. As per https://github.com/dotnet/runtime/pull/68874#discussion_r885010053 this should be no longer needed, so I can ideally just remove it in the follow up PR

Reply:
Logged https://github.com/dotnet/runtime/issues/70052